### PR TITLE
Feature/use current stable nginx version

### PR DESCRIPTION
**Pull request by syerad**
Debian is using the nginx version 1.14.2, which includes several security issues and is missing features which are already included in newer  version.

Therefore these changes included in this PR will add the official Nginx repository to the debian sources in order to fetch the newest stable version.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -17,7 +17,7 @@ full:      provision build
 all:       build
 
 build:
-	python ./bin/console docker:build --threads=auto
+	python2 ./bin/console docker:build --threads=auto
```
Comment (syerad) [inline Makefile:20]: for me the ```python```command is not working in the build container.

File: template/Dockerfile/images/nginx.jinja2 [hunk 2]
```diff
@@ -14,11 +14,14 @@
         nginx
 {%- endmacro %}
 
-
-{% macro debian(role='') -%}
+{% macro debian(role='',osCodename='buster') -%}
     # Install nginx
+    && printf "deb http://nginx.org/packages/debian/ {{ osCodename }}  nginx\n deb-src http://nginx.org/packages/debian/ {{ osCodename }} nginx" \
```
Comment (hhoechtl) [inline template/Dockerfile/images/nginx.jinja2:19]: You don't need to manually write the codename. You can do `$(docker-image-info dist-codename)`. This way you don't need to introduce new arguments to the macro.

File: template/Dockerfile/images/nginx.jinja2 [hunk 3]
```diff
@@ -14,11 +14,14 @@
...
+    >> /etc/apt/sources.list \
+    && curl -L https://nginx.org/keys/nginx_signing.key | apt-key add - \
+    && apt-get update \
```
Comment (hhoechtl) [inline template/Dockerfile/images/nginx.jinja2:22]: You don't need `apt-get update` our `apt-install` command already includes that. See https://github.com/webdevops/Docker-Image-Baselayout/blob/master/root/usr/local/bin/apt-install

**Review by hhoechtl — CHANGES_REQUESTED** [review]
In general please also commit the already generated templates `make provision` which generates the new Dockerfiles in `/docker`. It's not really necessary but we had it in the past, that jinja had some issues with newlines or blanklines in the final Dockerfile.

File: template/Dockerfile/images/nginx.jinja2 (hunk 3 above)
Comment (syerad) [reply template/Dockerfile/images/nginx.jinja2:22]: Okay thanks, didn't see that!

File: template/Dockerfile/images/nginx.jinja2 (hunk 2 above)
Comment (syerad) [reply template/Dockerfile/images/nginx.jinja2:19]: Will change that !

**Comment by tractorcow** [comment]
Thanks @syerad this helps me out a lot!

**Comment by syerad** [comment]
@jrencz 
I will take a look!
Give me a couple of days :)

**Comment by jrencz** [comment]
As I wrote in a comment there: I can just always express the very RUN, that you adjusted, right in my own Dockerfile to get latest nginx in a project that (for any reason) can't be easily upgraded to a still maintainined version

Right now I just want anyone who will - as I did - find this PR and think "so basically, I need to rebuild and we're done", that it will be like that but just for tags that were pushed after this was merged. I took what I saw in here overly optimistic :)

It may be that those tags that were altered in this PR would at some point get re-pushed to Dockerhub (who knows, some really super-important security update?). Then, and only then changes from this PR will be effective

**Comment by syerad** [comment]
@jrencz okay, understand.
As @hhoechtl mentioned in his comment, the build process for EOL versions for PHP is done manually.
Hence you need to ask Hans if he would like to push an updated version of an older image as well.

**Pull request merged**